core/src/utils/prove.rs

```diff
@@ -89,9 +86,123 @@ pub fn run_test(program: Program) -> Result<(), crate::stark::ProgramVerificatio
     Ok(())
 }
 
-pub fn prove_elf(elf: &[u8]) -> crate::stark::Proof<BabyBearBlake3> {
-    let program = Program::from(elf);
-    prove(program)
+fn trace_checkpoint(program: Program, file: &File) -> ExecutionRecord {
```

Comment:
It feels like this method should actually be impl'd on Runtime. It's also weird that the "file-handling" is mixed with the actual logic.

What do you think of:

let runtime = Runtime::recover(program, state);
runtime.execute_record();

Reply:
yea that seems good